src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2826,33 +2826,29 @@ protected function asDateTime($value)
         // these checks since they will be a waste of time, and hinder performance
         // when checking the field. We will just return the DateTime right away.
         if ($value instanceof DateTime) {
```

Comment:
There is no need for that as if `$value` is an instance of Carbon it's also an instance of DateTime so will be caught by the first check.